src/python/pants/bin/pants_exe.py

```diff
@@ -31,10 +32,12 @@ def main():
   start_time = time.time()
 
   exiter = Exiter()
-  exiter.set_except_hook()
+  ExceptionSink.set_exiter(exiter)
 
   with maybe_profiled(os.environ.get('PANTSC_PROFILE')):
     try:
       PantsRunner(exiter, start_time=start_time).run()
     except KeyboardInterrupt:
+      # TODO: ExceptionSink could probably take care of this too, as well as converting SIGINT into
```

Comment:
Not sure about that one. Where possible, I feel like treating the `Exiter` as a high level concept associated with a particular run, and `ExceptionSink` as the "always available manager of exception logs (and maybe an err stream if it's available)" is a cleanish separation.

With regard to handling keyboard interrupts: those are pretty clearly a per-run concept, and so if possible, their Handling should be part of the `Exiter`.

Reply:
Hm, ok, defining it like that I think is better than what I had in my head, which was "`ExceptionSink` makes sure your logs are logged" (subsuming trapping as well as logging -- the first half of "ExceptionSink"). If we can move global state e.g. changing signal handlers to the Exiter, or to its file, I think that's a great move.